File: python-backend/app/models/server_log.py
"""ServerLog ORM model."""

from sqlalchemy import BigInteger, Index, Integer, String
from sqlalchemy.orm import Mapped, mapped_column

from app.core.database import Base


class ServerLog(Base):
    __tablename__ = "server_logs"

    id: Mapped[int] = mapped_column(Integer, primary_key=True, autoincrement=True)
    type: Mapped[str] = mapped_column(String, nullable=False)
    message: Mapped[str] = mapped_column(String, nullable=False)
    error_stack: Mapped[str | None] = mapped_column(String, nullable=True)
    timestamp: Mapped[int] = mapped_column(BigInteger, nullable=False)
    created_at: Mapped[int] = mapped_column(BigInteger, nullable=False)

    __table_args__ = (
        Index("idx_server_log_type_time", "type", "timestamp"),
        Index("idx_server_log_time", "timestamp"),
    )

File: golang-backend/internal/model/server_log.go
package model

type ServerLog struct {
	ID         int64  `gorm:"primaryKey;autoIncrement" json:"id"`
	Type       string `gorm:"not null;index:idx_server_log_type_time" json:"type"`
	Message    string `gorm:"not null" json:"message"`
	ErrorStack *string `gorm:"column:error_stack" json:"errorStack"`
	Timestamp  int64  `gorm:"not null;index:idx_server_log_type_time;index:idx_server_log_time" json:"timestamp"`
	CreatedAt  int64  `gorm:"column:created_at" json:"createdAt"`
}

func (ServerLog) TableName() string { return "server_logs" }
